addons/spreadsheet/static/src/global_filters/components/filter_value/filter_value.xml

```diff
@@ -22,6 +22,25 @@
                 <DateFilterValue value="filterValue"
                                  update.bind="(value) => this.onDateInput(filter.id, value)"/>
             </div>
+            <div t-if="filter.type === 'numeric'" class="w-100 d-flex align-items-center"
+                 t-att-class="filterValue?.operator === 'between' ? 'justify-content-between' : ''">
+                <NumericFilterValue
+                    value="filterValue.targetValue"
+                    onValueChanged="(value) => this.onTargetValueNumericInput(filter.id, value)"
+                    t-if="filterValue?.operator !== 'between'"
+                />
+                <NumericFilterValue
+                    value="filterValue.minimumValue"
+                    onValueChanged="(value) => this.onMinimumValueNumericInput(filter.id, value)"
+                    t-if="filterValue?.operator === 'between'"
+                />
+                <i class="fa fa-long-arrow-right mx-2" t-if="filterValue?.operator === 'between'"/>
+                <NumericFilterValue
+                    value="filterValue.maximumValue"
+                    onValueChanged="(value) => this.onMaximumValueNumericInput(filter.id, value)"
+                    t-if="filterValue?.operator === 'between'"
+                />
```

Comment:
you can wrap those 3 elements inside a `<t t-if="filterValue?.operator === 'between'">` to avoid repeating the if condition 3 times.
You can also move the first component with `t-if="filterValue?.operator !== 'between'"` below with a `t-else=""`